### PR TITLE
Exception 2

### PR DESCRIPTION
**Pull request by awalsh3**
When the serializer does not return a JSON string but, instead raises an exception.

### CODE REVIEW

File: app/controllers/api/v1/books_controller.rb [hunk 1]
```diff
@@ -2,38 +2,29 @@ module Api
   module V1
     class BooksController < ApplicationController
       def index
-        books = Book.all
+        raise StandardError, "Unable to serialize books" if serialized_books.nil?
```
Comment (patrickquigley102) [inline app/controllers/api/v1/books_controller.rb:5]: You've gone down the wrong path here.

But it's very understandable why you have, as we went down that path in Exceptions 1.

Here, you shouldn't be making any changes to the actual code. Only the test code. 

We'll use RSpec's mocking behaviour to mock an exception being raised.

Not actually raise an exception in the real code.

We can book in some extra time to pair on this if you like? Let me know.

Comment (awalsh3) [reply app/controllers/api/v1/books_controller.rb:5]: That would be great, thanks! Can you still meet tomorrow or shall we meet up on 28/7?

Comment (patrickquigley102) [reply app/controllers/api/v1/books_controller.rb:5]: I've sent more detail in an email but, yes the 28th sounds good :)

**Comment by patrickquigley102** [comment]
I think we can close this one pal, it's served it's use. Do you agree?

**Pull request closed without merging**